src/V3Randomize.cpp

```diff
@@ -473,6 +463,14 @@ class RandomizeMarkVisitor final : public VNVisitor {
         iterateChildrenConst(nodep);
     }
 
+    void visit(AstNodeExpr* nodep) override {
+        iterateChildrenConst(nodep);
+        if (!m_constraintExprGenp) return;
+        nodep->user1((nodep->op1p() && nodep->op1p()->user1())
+                     || (nodep->op2p() && nodep->op2p()->user1())
+                     || (nodep->op3p() && nodep->op3p()->user1())
+                     || (nodep->op4p() && nodep->op4p()->user1()));
```

Comment:
If we later add some specific visitor that subclasses AstNodeExpr this will break.  Also causes a lot of cache polution as user1 is written. Is there a way to use a visitor member variable instead to trigger the warning?

Reply:
I set user1 not only to trigger the warning, but also because other visitors checks it. Currently on master it isn't set correctly and I think that some cases may be handled wrong because of it.

Reply:
So I think that `user1` has to be set in all expressions. I think that no matter in which way we will decide to set it, we will have to do it in all visitors of subclasses of `AstNodeExpr`.